Content.Server/Fax/FaxSystem.cs

```diff
@@ -0,0 +1,390 @@
+using Content.Server.Paper;
+using Content.Server.Popups;
+using Content.Server.Tools;
+using Content.Server.UserInterface;
+using Content.Shared.Containers.ItemSlots;
+using Content.Shared.Emag.Systems;
+using Content.Shared.Fax;
+using Content.Shared.Interaction;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+using Robust.Shared.Player;
+
+namespace Content.Server.Fax;
+
+public sealed class FaxSystem : EntitySystem
+{
+    [Dependency] private readonly IChatManager _chat = default!;
+    [Dependency] private readonly IAdminManager _adminManager = default!;
+    [Dependency] private readonly ItemSlotsSystem _itemSlotsSystem = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearanceSystem = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+    [Dependency] private readonly PaperSystem _paperSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly ToolSystem _toolSystem = default!;
+    [Dependency] private readonly QuickDialogSystem _quickDialog = default!;
+    [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
+
+    public const string PaperSlotId = "Paper";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        
+        // Hooks
+        SubscribeLocalEvent<FaxMachineComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<FaxMachineComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<FaxMachineComponent, ComponentRemove>(OnComponentRemove);
+        
+        SubscribeLocalEvent<FaxMachineComponent, EntInsertedIntoContainerMessage>(OnItemSlotChanged);
+        SubscribeLocalEvent<FaxMachineComponent, EntRemovedFromContainerMessage>(OnItemSlotChanged);
+        SubscribeLocalEvent<FaxMachineComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+        
+        // Interaction
+        SubscribeLocalEvent<FaxMachineComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<FaxMachineComponent, GotEmaggedEvent>(OnEmagged);
+        
+        // UI
+        SubscribeLocalEvent<FaxMachineComponent, AfterActivatableUIOpenEvent>(OnToggleInterface);
+        SubscribeLocalEvent<FaxMachineComponent, FaxSendMessage>(OnSendButtonPressed);
+        SubscribeLocalEvent<FaxMachineComponent, FaxRefreshMessage>(OnRefreshButtonPressed);
+        SubscribeLocalEvent<FaxMachineComponent, FaxDestinationMessage>(OnDestinationSelected);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var comp in EntityQuery<FaxMachineComponent>())
+        {
+            // Printing animation
+            if (comp.PrintingTimeRemaining > 0)
+            {
+                comp.PrintingTimeRemaining -= frameTime;
+                UpdateAppearance(comp.Owner, comp);
```

Comment:
this should probably check if the device is currently powered.

Reply:
But what about the animation, can I somehow pause it in current frame and then continue when power return?

Reply:
Make interrupt of animation and reset to normal fax state